### PR TITLE
added desired capacity type flag in asg

### LINKED ISSUE #26641: desiredcapacityType seems to be missing
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Multiple resources mention setting the `desiredcapacityType`, but there's no parameter for it anywhere.
I also can't find it in the code of the autoscaling, for example.

### New or Affected Resource(s)

* aws_autoscaling_group
* aws_launch_template

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

* #0000

### PR DESCRIPTION
**Pull request by ajayOO8**
### Description
added flag for desired_capacity_type in resource aws auto-scaling-group

### Relations
closes #26641 

### Output from Acceptance Testing
$make testacc TESTS=TestAccAutoScalingGroupsDataSource_basic  PKG=autoscaling
```
 TF_ACC=1 /nail/home/ajayprataps/tools/go/bin/go test ./internal/service/autoscaling/... -v -count 1 -parallel 20 -run='TestAccAutoScalingGroupsDataSource_basic'  -tieout 180=== RUN   TestAccAutoScalingGroupsDat=== PAUSE TestAccAutoScalingGroupsDataSource_basic
=== CONT  TestAccAutoScalingGroupsDataSource_basic
--- PASS: TestAccAutoScalingGroupsDataSource_basic (35.32s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/autoscaling	35.432s
```

### CODE REVIEW

File: .changelog/28658.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+```release-note:enhancement
+resource/aws_asg: Add desired_capacity_type flag
```
Comment (bschaatsbergen) [inline .changelog/28658.txt:2]: `desired_capacity_type` attribute

File: internal/service/autoscaling/group.go [hunk 2]
```diff
@@ -81,6 +81,10 @@ func ResourceGroup() *schema.Resource {
 				Optional: true,
 				Computed: true,
 			},
+			"desired_capacity_type": {
+				Type:     schema.TypeString,
```
Comment (bschaatsbergen) [inline internal/service/autoscaling/group.go:85]: `Computed: true`
As you mentioned in the docs that there's a default value for it.

**Review by bschaatsbergen — CHANGES_REQUESTED** [review]
Added 2 drive-by comments, could you please also create a test for this new attribute?

File: website/docs/r/autoscaling_group.html.markdown [hunk 3]
```diff
@@ -405,6 +405,9 @@ The following arguments are supported:
 * `desired_capacity` - (Optional) Number of Amazon EC2 instances that
     should be running in the group. (See also [Waiting for
     Capacity](#waiting-for-capacity) below.)
+* `desired_capacity_type` - (Optional) specify the desired capacity type
```
Comment (bschaatsbergen) [inline website/docs/r/autoscaling_group.html.markdown:408]: See the `markdown-lint` Github Action that reported a linting issue please.

**Comment by ewbankkit** [comment]
@ajayOO8 Thanks for the contribution :tada: :clap:.
Everything was great.
In order to get this released today I went ahead and added some acceptance tests.

**Pull request merged**